test/nlp/test_default.py

```diff
@@ -57,15 +57,23 @@ def test_starting_point_not_in_search_space():
             "learning_rate": {
                 "domain": tune.choice([1e-4, 1e-5]),
             },
+            "per_device_train_batch_size": {
+                "domain": 2,
+            },
         }
     }
     automl_settings["starting_points"] = "data:test/nlp/default/"
     del automl_settings["fit_kwargs_by_estimator"][this_estimator_name]["model_path"]
 
     automl.fit(X_train, y_train, **automl_settings)
     assert (
-        len(automl._search_states[this_estimator_name].init_config) == 0
-    )  # check that init config is not updated, but search space is updated
+        len(automl._search_states[this_estimator_name].init_config)
+        == len(automl._search_states[this_estimator_name]._search_space_domain) - 3
+    ), (
+        "The search space is updated with the custom_hp on 3 hyperparameters of "
+        "the specified estimator without an initial value. Thus a valid init config "
+        "should only contain the cardinality of the search space minus 3."
```

Comment:
```suggestion
        "should only contain the cardinality of the search space minus {}".format(len(automl_settings["custom_hp"]),  len(automl_settings["custom_hp"]))
```

Reply:
`len(automl_settings["custom_hp"]` should be `len(automl_settings["custom_hp"][this_estimator_name]` ?